openedx/frontend-app-authoring
# Use soft nav when clicking a library from studio home [FC-0062]

## Description

Currently, clicking a library link as shown will cause a full page reload. This PR fixes that so it uses a react-router "soft nav" and just loads the minimum changes necessary to display the library. This makes the UI feel much faster and more responsive.

![Screenshot](https://github.com/user-attachments/assets/9405b9bd-5b75-45b2-8318-047f6b4d82da)


I **converted a lot of things to TypeScript** at the same time.

## Supporting information

This is a drive-by fix while working on [Private ref: [FAL-3827](https://tasks.opencraft.com/browse/FAL-3827)].

## Testing instructions

Test the **course list** and **library list** on the Studio hompage, and make sure they're working as before. Test the course list with both the `ENABLE_HOME_PAGE_COURSE_API_V2` setting as `true` and as `false` and make sure both are working.

Test that when clicking on a library, it loads fast and is not reloading the whole page.
